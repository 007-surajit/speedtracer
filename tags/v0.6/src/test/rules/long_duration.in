hintlet_long_duration.js

# Input Record 1
# Expected: none
# Shouldn't trigger a rule - too short.
{"children":[{"children":[],"data":{},"duration":0.9620,"time":5088.605999998748,"type":2}],"data":{"timerId":"0TimerEvent28"},"duration":1.024,"time":1.,"type":12}

# Input Record 2
# Expected: HINTLET RECORD
# Should trigger a rule w/ warning severity.
{"children":[{"children":[],"data":{},"duration":50.9,"time":5088.605999998748,"type":2}],"data":{"timerId":"0TimerEvent28"},"duration":500.9,"time":2.,"type":12}

# Input Record 3
# Expected: HINTLET RECORD
# Should trigger a critical alarm.
{"children":[{"children":[],"data":{},"duration":30.9,"time":5088.605999998748,"type":2}],"data":{"timerId":"0TimerEvent28"},"duration":3000.2,"time":3.,"type":12}

# Input Record 4
# Expected: none
# Shouldn't trigger a rule.
# Included because it doesn't contain a duration field.
{"data":{"interval":100.0,"singleShot":true,"timerId":"0TimerEvent31"},"time":5089.548000000417,"type":13}
