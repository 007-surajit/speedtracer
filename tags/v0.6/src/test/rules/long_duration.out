{"hintletRule":"Long Duration Events","timestamp":2,"description":"Event lasted: 501ms.  Exceeded threshold: 100ms","refRecord":2,"severity":3}
{"hintletRule":"Long Duration Events","timestamp":3,"description":"Event lasted: 3000ms.  Exceeded threshold: 2000ms","refRecord":3,"severity":2}
